src/puppetlabs/puppetdb/http/command.clj

```diff
@@ -67,11 +67,24 @@
   (with-chan [response-chan (async/chan)]
     (with-sub {:from response-pub :topic (str uuid) :chan response-chan}
       (let [timeout-chan (async/timeout completion-timeout-ms)
-            _ (do-submit-fn)
-            timed-out (async/alt!! timeout-chan true
-                                   response-chan false)]
-        (http/json-response {:uuid uuid, :processed (not timed-out)}
-                            (if timed-out 503 200))))))
+            _ (do-submit-fn)]
+        (async/alt!!
+          timeout-chan (http/json-response {:uuid uuid
+                                            :processed false
+                                            :timed_out true}
+                                           503)
+          response-chan ([{:keys [command exception]}]
+                         (if exception
+                           (http/json-response {:uuid uuid
+                                                :processed true
+                                                :timed_out false
+                                                :error (str exception)
+                                                :stack_trace (map str (.getStackTrace exception))}
+                                               503)
+                           (http/json-response {:uuid uuid
+                                                :processed true
+                                                :timed_out false}
+                                               200))))))))
```

Comment:
Maybe the base response could get named and put in a let, and for the error case the :error and :stack_trace keys could be assoc'd in or merged in